darwin/dataset/download_manager.py

```diff
@@ -95,14 +97,14 @@ def download_all_images_from_annotations(
 
         if not force_replace:
             # Check the planned path for the image against the existing images
-            filename = Path(annotation.filename)
-            if use_folders and annotation.remote_path != "/":
-                planned_image_path = (
-                    images_path / Path(annotation.remote_path.lstrip("/\\"))
-                ) / filename
-            else:
-                planned_image_path = images_path / filename
-            if planned_image_path in existing_images:
+            planned_image_paths = _get_planned_image_paths(
+                annotation, images_path, use_folders
+            )
+
+            if all(
+                planned_image_path in existing_images
+                for planned_image_path in planned_image_paths
+            ):
```

Comment:
We want `if all` here because we should re-download a multi-slotted file if any local source file is missing